pennylane/transforms/optimization/single_qubit_fusion.py

```
@@ -14,19 +14,240 @@
+                self._env[invar] = arg
+            for const, constvar in zip(consts, jaxpr.constvars, strict=True):
+                self._env[constvar] = const
+
+            for eqn in jaxpr.eqns:
+
+                prim_type = getattr(eqn.primitive, "prim_type", "")
+
+                custom_handler = self._primitive_registrations.get(eqn.primitive, None)
+                if custom_handler:
+                    self.interpret_all_previous_ops()
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    outvals = custom_handler(self, *invals, **eqn.params)
+                elif prim_type == "operator":
+                    outvals = self.interpret_operation_eqn(eqn)
+                elif prim_type == "measurement":
+                    self.interpret_all_previous_ops()
+                    outvals = self.interpret_measurement_eqn(eqn)
+                else:
+                    if prim_type == "transform":
+                        self.interpret_all_previous_ops()
+                    invals = [self.read(invar) for invar in eqn.invars]
+                    subfuns, params = eqn.primitive.get_bind_params(eqn.params)
+                    outvals = eqn.primitive.bind(*subfuns, *invals, **params)
+
+                if not eqn.primitive.multiple_results:
+                    outvals = [outvals]
+                for outvar, outval in zip(eqn.outvars, outvals, strict=True):
+                    self._env[outvar] = outval
+
+            self.interpret_all_previous_ops()
+
+            outvals = []
+            for var in jaxpr.outvars:
+                outval = self.read(var)
+                if isinstance(outval, Operator):
+                    outvals.append(super().interpret_operation(outval))
+                else:
+                    outvals.append(outval)
+
+            self.cleanup()
+            return outvals
+
+    def single_qubit_fusion_plxpr_to_plxpr(
+        jaxpr, consts, targs, tkwargs, *args
```

Compared to a standard Python function, this will silently ignore `tkwargs` that are not `"atol"` or `"exclude_gates"`, right? Is this the non-plxpr behaviour as well? Do we want to make sure not to silently discard user inputs?

---

Good point. We should also update this for all the other transforms that have been captured (see [this](https://github.com/PennyLaneAI/pennylane/blob/7c239b6337ef957d2caaad1789159df49fb626fb/pennylane/transforms/decompose.py#L290) as an example).

Should we raise an error if the user provides invalid arguments for the transform? We can also postpone the discussion until we have a better integration with transforms and program capture. Happy to create a story and discuss this if needed